data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/PipelineConnector.java

```diff
@@ -54,6 +56,21 @@ public void stop() {
     public void output(final Collection<T> records) {
         if (buffer != null && !isStopRequested.get()) {
             for (T record : records) {
+		if(record.getData() instanceof Event) {
+		    try {
+		        final Event recordEvent = (Event)record.getData();
+			Event newRecordEvent = JacksonEvent.builder() 
```

Comment:
This will not work for Trace events. Please take a similar approach to the following here:

https://github.com/opensearch-project/data-prepper/blob/7901643baccc5e75675eb578362e2ea587a5ca9a/data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/server/PeerForwarderHttpService.java#L117-L127